core/drivers/firewall/stm32_risab.c

```diff
@@ -0,0 +1,813 @@
+
+	assert((unsigned int)(lenp / sizeof(uint32_t)) == 2);
+
+	risab_d->region_cfged.base = fdt32_to_cpu(cuint[0]);
+	risab_d->region_cfged.size = fdt32_to_cpu(cuint[1]);
+
+	/* Get the memory regions to configure */
+	mem_regions = fdt_getprop(fdt, node, "memory-region", &lenp);
+	if (!mem_regions)
+		panic("No memory region to configure");
+
+	risab_d->nb_regions_cfged = (unsigned int)(lenp / sizeof(uint32_t));
+	assert(risab_d->nb_regions_cfged < _RISAB_NB_PAGES_MAX);
+
+	risab_d->subr_cfg = calloc(risab_d->nb_regions_cfged,
+				   sizeof(*risab_d->subr_cfg));
+	if (!risab_d->subr_cfg)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	for (i = 0; i < risab_d->nb_regions_cfged; i++) {
+		uint32_t phandle = fdt32_to_cpu(mem_regions[i]);
+		size_t sub_region_offset = 0;
+		paddr_t address = 0;
+		size_t length = 0;
+
+		mem_reg_node = fdt_node_offset_by_phandle(fdt, phandle);
+		if (mem_reg_node < 0)
+			return TEE_ERROR_ITEM_NOT_FOUND;
+
+		/*
+		 * Get the reg property to determine the number of pages
+		 * to configure
+		 */
+		address = fdt_reg_base_address(fdt, mem_reg_node);
+		length = fdt_reg_size(fdt, mem_reg_node);
+
+		assert(IS_ALIGNED(address, _RISAB_PAGE_SIZE) &&
+		       IS_ALIGNED(length, _RISAB_PAGE_SIZE));
+
+		/*
+		 * Get the sub region offset and check if it is not out
+		 * of bonds
+		 */
+		sub_region_offset = address - risab_d->region_cfged.base;
+
+		if (!core_is_buffer_inside(address, length,
+					   risab_d->region_cfged.base,
+					   risab_d->region_cfged.size)) {
+			EMSG("Region %#"PRIxPA"..%#"PRIxPA" outside RISAB area %#"PRIxPA"...%#"PRIxPA,
+			     address, address + length,
+			     risab_d->region_cfged.base,
+			     risab_d->region_cfged.base +
+			     risab_d->region_cfged.size);
+			return TEE_ERROR_BAD_PARAMETERS;
+		}
+
+		risab_d->subr_cfg[i].first_page = sub_region_offset /
+						  _RISAB_PAGE_SIZE;
+		risab_d->subr_cfg[i].nb_pages_cfged = length /
+						      _RISAB_PAGE_SIZE;
+		if (!risab_d->subr_cfg[i].nb_pages_cfged)
+			panic("Range to configure is < to the size of a page");
+
+		/* Get the RIF configuration for this region */
+		cuint = fdt_getprop(fdt, mem_reg_node, "st,protreg", &lenp);
+		if (!cuint)
+			panic("No RIF configuration available");
+
+		/* There should be only one configuration for this region */
+		assert((unsigned int)(lenp / sizeof(uint32_t)) == 1);
+
+		parse_risab_rif_conf(risab_d, &risab_d->subr_cfg[i],
+				     fdt32_to_cpu(cuint[0]),
+				     true /*check_overlap*/);
+	}
+
+	return TEE_SUCCESS;
+}
+
+static void enable_srwiad_if_set(struct stm32_risab_pdata *risab_d)
+{
+	if (is_tdcid && risab_d->srwiad)
+		io_setbits32(risab_base(risab_d), _RISAB_CR_SRWIAD);
+};
+
+static void disable_srwiad_if_unset(struct stm32_risab_pdata *risab_d)
+{
+	if (is_tdcid && !risab_d->srwiad)
+		io_clrbits32(risab_base(risab_d), _RISAB_CR_SRWIAD);
+};
+
+static void clear_iac_regs(struct stm32_risab_pdata *risab_d)
+{
+	io_setbits32(risab_base(risab_d) + _RISAB_IACR, GENMASK_32(1, 0));
```

Comment:
For consistency, I suggest to replace `GENMASK_32(1, 0)` with `_RISAB_IACR_CAEF | _RISAB_IACR_IAEF`
or define the mask above:
```patch
 /* RISAB_IACR bitfields */
 #define _RISAB_IACR_CAEF			BIT(0)
 #define _RISAB_IACR_IAEF			BIT(1)
+#define _RISAB_IACR_MASK			(_RISAB_IACR_CAEF |
+						 _RISAB_IACR_IAEF)